docs/docs/miscellaneous/hooks.mdx

```diff
@@ -0,0 +1,32 @@
+# Hooks
+
+Elf allows customizing some of the behavior via `elfHooks`
+
+## registerPreStoreUpdate
+
+Provide a function to modify the store value after reducers has run.
+
+The callback function takes three arguments:
+
+- `currentState` - state before reducers has run
+- `nextState` - state after reducers has run
+- `storeName` - the name of the store
+
+One scenario when this functionality could be useful is when a developer wants to prevent bugs associated with mutability by freezing the store value during the development:
+
+```ts
+import { elfHooks, deepFreeze } from '@ngneat/elf';
+
+if (!environment.production) {
+  elfHooks.registerPreStoreUpdate<typeof store>(
```

Comment:
Now I'm not sure the the generic makes sense here because this is a global hook, not local.

Reply:
yeah, for the purposes of `freeze` the types aren't really needed. I'm thinking of the use-case when the developer would actually want to perform some operation on a particular store. In that case, having type-safety would be good.
Better to have an option to specify the type of the store you're working on than not?

Reply:
I don't see the point If it's a global hook

Reply:
sounds good. I'll remove it